python/tank_vendor/shotgun_authentication/session_cache.py

```diff
@@ -385,25 +386,32 @@ def generate_session_token(hostname, login, password, http_proxy):
     :param login: The user to get a session for.
     :param password: Password for the user.
     :param http_proxy: Proxy to use. Can be None.
+    :param auth_token: Two factor authentication token for the user. Can be None.
 
-    :returns: The generated session token for that user/password/site combo.
+    :returns: The generated session token for that user/password/site combo. If the login
```

Comment:
I tend to reserve exceptions for stuff that indicate an input error (bad password, invalid site) or some unforeseen issue, which requiring 2fa is none of. Note that if you enter a bad 2fa code however, it will throw.